src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ReaderMetricsProcessor.java

```diff
@@ -71,6 +82,11 @@
   private final AppContext appContext;
   private final ConfigOverridesApplier configOverridesApplier;
 
+  public static final String BATCH_METRICS_ENABLED_CONF_FILE = "batch_metrics_enabled.conf";
+  private boolean batchMetricsEnabled;
+  private boolean defaultBatchMetricsEnabled = false;
+  private ConcurrentSkipListSet<Long> batchMetricsDBSet;
```

Comment:
Let's add a comment here saying that this datastructure can be modified by the writer/cleanup thread while being read by the other thread trying to batch metrics. And therefore, it needs to be `ConcurrentSkipListSet`

Reply:
Sure